store/cachekv/store.go

```diff
@@ -177,18 +179,48 @@ func (store *Store) iterator(start, end []byte, ascending bool) types.Iterator {
 	return newCacheMergeIterator(parent, cache, ascending)
 }
 
+// strToByte is meant to make a zero allocation conversion
+// from string -> []byte to speed up operations, it is not meant
+// to be used generally, but for a specific pattern to check for available
+// keys within a domain.
+func strToByte(s string) []byte {
```

Comment:
Mostly memo to self: there is another implementation in `types/address/hash.go`. We'll eventually need to 1. find a place where to put these functions (and I hope we won't come up with another catch-all `util` package) 2. reduce code duplication

Reply:
Ack, we can do so later indeed, but for now this'll suffice and keep the PR focused :-)

Reply:
Completely agree! Thanks for the great work! :pray:

Reply:
Great job @odeke-em. Are you planning to merge str/bytes convertion functions? Otherwise I can do a quick PR for that.

Reply:
Thanks @robert-zaremba! Please go ahead and add them, but perhaps we should put them in an internal package so that no other packages out of the cosmos-sdk can import them.

Reply:
yes, :+1: